boto3/compat.py

```diff
@@ -16,22 +16,13 @@
 import socket
 import warnings
 
-from botocore.vendored import six
 from boto3.exceptions import PythonDeprecationWarning
 
-if six.PY3:
-    # In python3, socket.error is OSError, which is too general
-    # for what we want (i.e FileNotFoundError is a subclass of OSError).
-    # In py3 all the socket related errors are in a newly created
-    # ConnectionError
-    SOCKET_ERROR = ConnectionError
-else:
-    SOCKET_ERROR = socket.error
-
-if six.PY3:
-    import collections.abc as collections_abc
-else:
-    import collections as collections_abc
```

Comment:
I don't think we can delete this entirely. We need to keep `import collections.abc as collections_abc`.

Reply:
To be clear, we should also keep the compat references in other modules so this doesn't need to be updated in several places later if there's a new collections alternative.